### PR TITLE
fix(import): Do not import search values when they are not provided

### PR DESCRIPTION
**Pull request by junajan**

### CODE REVIEW

File: src/coffee/mapping.coffee [hunk 1]
```diff
@@ -367,6 +367,8 @@ class Mapping
     if _.has langH2i, attribName
       _.each langH2i[attribName], (index, language) ->
         val = row[index]
+        if val == ""
```
Comment (hisabimbola) [inline src/coffee/mapping.coffee]: Any reason why you specifically checking for empty strings?

Better is you check for falsy values?

Comment (junajan) [inline src/coffee/mapping.coffee]: I am translating null/undefined to an empty string. Also, the problem was that an empty string was not removed from searchKeywords (empty string is not allowed there)

File: src/coffee/io/reader.coffee [hunk 2]
```diff
@@ -78,7 +78,9 @@ class Reader
         rowValues = row.values
         rowValues.shift()
         rows.push _.map rowValues, (item) ->
-          if item then String(item) else ""
+          if typeof(item) == "undefined" || item == null
```
Comment (hisabimbola) [inline src/coffee/io/reader.coffee]: Can header by Number `0`?

File: src/coffee/io/reader.coffee [hunk 3]
```diff
@@ -77,7 +77,10 @@ class Reader
       worksheet.eachRow (row) =>
         rowValues = row.values
         rowValues.shift()
-        rows.push rowValues.map String
+        rows.push _.map rowValues, (item) ->
+          if typeof(item) == "undefined" || item == null
```
Comment (hisabimbola) [inline src/coffee/io/reader.coffee]: In coffeescript, will using `?` work.

```coffeescript
if item?
  # do something
```

Comment (junajan) [inline src/coffee/io/reader.coffee]: @hisabimbola will it skip 0 as well?

**Comment by junajan** [comment]
@hisabimbola test added here 479f9de

File: src/coffee/io/reader.coffee (hunk 3 above)
Comment (junajan) [inline src/coffee/io/reader.coffee]: @hisabimbola updated here 5850419

File: src/spec/mapping.spec.coffee [hunk 4]
```diff
@@ -221,6 +221,38 @@ describe 'Mapping', ->
         done()
       .catch done
 
+    it 'should map empty search keywords', (done) ->
```
Comment (hisabimbola) [inline src/spec/mapping.spec.coffee]: Can we refactor this test and call mapSearchKeyword directly.

and we test this use case.

Comment (hisabimbola) [inline src/spec/mapping.spec.coffee]: Or maybe add both.

File: src/coffee/mapping.coffee [hunk 5]
```diff
@@ -367,6 +367,8 @@ class Mapping
     if _.has langH2i, attribName
       _.each langH2i[attribName], (index, language) ->
         val = row[index]
+        if not _.isString(val) || val == ""
```
Comment (hisabimbola) [inline src/coffee/mapping.coffee]: Can you explain why we doing this?

What are the use case. if we have a predefined format for searchKeywords, if that format is not followed shouldn't we return an error.

Let's discuss about this.

File: src/spec/mapping.spec.coffee [hunk 6]
```diff
@@ -221,6 +221,38 @@ describe 'Mapping', ->
...
+      csv =
+      """
```
Comment (junajan) [inline src/spec/mapping.spec.coffee]: It is copied from another test :-(

**Pull request closed without merging**